cardinal/server/server_test.go

```diff
@@ -122,8 +117,51 @@ func setTestTimeout(t *testing.T, timeout time.Duration) {
 	}()
 }
 
+func TestHealthEndpoint(t *testing.T) {
+	//setTestTimeout(t, 10*time.Second)
+	w := inmem.NewECSWorldForTest(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	w.AddSystem(func(world *ecs.World, queue *transaction.TxQueue, _ *log.Logger) error {
+		return nil
+	})
+	assert.NilError(t, w.LoadGameState())
+	txh := makeTestTransactionHandler(t, w, DisableSignatureVerification())
+	resp, err := http.Get("http://localhost:4040/health")
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, 200)
+	body := make(map[string]bool)
+	err = json.NewDecoder(resp.Body).Decode(&body)
+	assert.NilError(t, err)
+	v, ok := body["is_server_running"]
+	assert.Assert(t, ok)
+	assert.Assert(t, v)
+	v, ok = body["is_game_loop_running"]
+	assert.Assert(t, ok)
+	assert.Assert(t, !v) //game loop is not running!
+	ctx := context.Background()
+	w.StartGameLoop(ctx, time.Tick(1*time.Second), nil)
+	isGameLoopRunning := false
+	for !isGameLoopRunning {
```

Comment:
the test passes for me without this loop, do we actually need it?

Reply:
Yes. StartGameLoop does not block it launches a goroutine.